src/sayfudinova/task2/CubeWindow.cpp

```diff
@@ -0,0 +1,151 @@
+#include "CubeWindow.h"
+
+#include <QColorDialog>
+#include <QMouseEvent>
+#include <QOpenGLFunctions>
+#include <QScreen>
+
+struct VertexData{
+  VertexData(){ }
+  VertexData(QVector3D p) :
+      position(p){ }
+  QVector3D position;
+};
+
+void CubeWindow::initCube(float width, int N) {
+  float halfWidth = width / 2;
+  float step = width / float(N - 1);
+
+  QVector<VertexData> vertices;
+
+  for (auto z = -1; z <= 1; z += 2) {
+    for (auto y = 0; y < N; y++) {
+      for (auto x = 0; x < N; x++) {
+        vertices.append(VertexData(
+            QVector3D(z*(-halfWidth +  x * step), -halfWidth + y * step, z * halfWidth)));
+      }
+    }
+  }
+  for (auto x = -1; x <= 1; x += 2) {
+    for (auto z = 0; z < N; z++) {
+      for (auto y = 0; y < N; y++) {
+        vertices.append(VertexData(
+            QVector3D(x * halfWidth, -halfWidth + y * step, x*(-halfWidth + z * step))));
+      }
+    }
+  }
+  for (auto y = -1; y <= 1; y += 2) {
+    for (auto x = 0; x < N; x++) {
+      for (auto z = 0; z < N; z++) {
+        vertices.append(VertexData(
+            QVector3D(-halfWidth + x * step, y * halfWidth, y*(-halfWidth + z * step))));
+      }
+    }
+  }
+
+
+  QVector<GLuint> indices;
+  for (int i = 0; i < 6 * N * N; i += N * N) {
+    for (int j = 0; j < (N - 1) * (N - 1); j += N) {
+      for (int k = 0; k < (N - 1); k++) {
+        indices.append(i + j + k + N);
+        indices.append(i + j + k + 0);
+        indices.append(i + j + k + N + 1);
+        indices.append(i + j + k + N + 1);
+        indices.append(i + j + k + 0);
+        indices.append(i + j + k + 1);
+      }
+    }
+  }
+
+  vbo.create();
+  vbo.bind();
+  vbo.allocate(vertices.constData(), vertices.size() * sizeof(VertexData));
+
+  ibo.create();
+  ibo.bind();
+  ibo.allocate(indices.constData(), indices.size() * sizeof(GLuint));
+}
+
+void CubeWindow::init() {
+  // Enable depth test and face culling
+  glEnable(GL_DEPTH_TEST);
+  glEnable(GL_CULL_FACE);
+
+  // Clear all buffers
+  glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+  start = std::chrono::system_clock::now();
```

Comment:
Давайте поле класса старт тоже напишем с суффиксом `_`, как и другие поля (для единообразия)
```suggestion
  start_ = std::chrono::system_clock::now();
```